### PR TITLE
Add CSV transform for traject

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

**Comment by cmh2166** [comment]
Will look at this / test tomorrow (Sunday).

File: lib/traject/macros/csv.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Macros
+  # Macros for extracting TEI values from Nokogiri documents
```
Comment (jcoyne) [inline lib/traject/macros/csv.rb]: TEI?

File: lib/traject/macros/csv.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  module Csv
+    # @param xpath [String] the xpath query expression
```
Comment (jcoyne) [inline lib/traject/macros/csv.rb]: This is incorrect documentation

File: lib/traject/macros/csv.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    def column(index_or_name, options = {})
+      lambda do |row, accumulator, _context|
+        result = Array(row[index_or_name])
+        result = result.map(&:strip) if options.key?(:trim)
```
Comment (jcoyne) [inline lib/traject/macros/csv.rb]: Trimming is not tested. Should we remove this line?

Comment (cbeer) [inline lib/traject/macros/csv.rb]: I think it's worth keeping, if only to ensure some consistency between our extraction options. I expect we'll soon be able to pull things like this and translation maps up into a more generic value filtering / mapping method?

**Pull request merged**